Outbound Routes: Uploaded csv-file is not deemed valid csv (Fixed)

hannes427 opened the pull request:
In special circumstances a uploaded csv-file with dial patterns is not recognized as valid, because some browsers send another mime-type. This PR removes the check if a csv-file was uploaded based on the mime type and adds a check if the uploaded file is in the correct format for FreePBX 17.

Fixed missing declaration of variables

Bugfix FreePBX/issue-tracker#693

hannes427 committed:
Outbound Routes: Uploaded csv-file is not deemed valid csv (Fixed)

In special circumstances a uploaded csv-file with dial patterns is not recognized as valid, because some browsers send another mime-type.
This PR removes the check if a csv-file was uploaded based on the mime type and adds a check if the uploaded file is in the correct format for FreePBX 17.

Fixed missing declaration of variables

Bugfix FreePBX/issue-tracker#693

miken32 commented:
If you're already reading the entire file in with `file()` why not replace the following `fopen()`/`fgetcsv()`/`fclose()` structure instead of reading the file twice? Also: if you `return` from an `if` block there's no need for an `else` after it, you should use strict equality checks, and you should match your code style to the existing code, e.g. `} else {`.

miken32 commented:
Also, the UI says "You **_can_** also supply headers: prepend, prefix, match pattern and callerid in the first row." The header row has never been required.

hannes427 commented:
@miken32 Updated PR with your suggestions

hannes427 committed:
Outbound Routes: Uploaded csv-file is not deemed valid csv (Fixed)
In special circumstances a uploaded csv-file with dial patterns is not recognized as valid, because some browsers send another mime-type. This PR removes the check if a csv-file was uploaded based on the mime type and adds a check if the uploaded file is in the correct format for FreePBX 17.

* Modified regex, so that the header is optional in the csv-file
* Removed fopen()/fgetcsv() of the uploaded file to prevent reading it
  twice
* Adjust code style

Bugfix FreePBX/issue-tracker#693
